crates/flowctl/src/raw/suggest_schema.rs

```diff
@@ -0,0 +1,222 @@
+use anyhow::Context;
+use bytes::Bytes;
+use doc::{inference::Shape, SchemaIndexBuilder, FailedValidation};
+use futures::{TryStreamExt, StreamExt};
+use journal_client::{broker, read::uncommitted::{ReadUntil, ReadStart, JournalRead, ExponentialBackoff, Reader}, list::list_journals, AuthHeader};
+use json::schema::{build::build_schema, types};
+use models::{
+    Capture, CaptureBinding, CaptureDef, CaptureEndpoint, Catalog, Collection, CollectionDef,
+    CompositeKey, ConnectorConfig, JsonPointer, Schema, ShardTemplate,
+};
+use proto_flow::{
+    capture::{request, Request},
+    flow::capture_spec::ConnectorType,
+    ops::Log
+};
+use bytelines::AsyncByteLines;
+use proto_grpc::broker::journal_client::JournalClient;
+use schema_inference::{json_decoder::JsonCodec, inference::infer_shape, shape, server::InferenceError, schema::SchemaBuilder};
+use serde_json::{json, value::RawValue, Value};
+use tokio::io::{BufReader, AsyncReadExt};
+use tokio_util::{compat::FuturesAsyncReadCompatExt, io::{ReaderStream, StreamReader}, codec::FramedRead};
+use tonic::{codegen::InterceptedService, transport::Channel};
+use std::{collections::BTreeMap, io::ErrorKind, time::{Duration, Instant}};
+use url::Url;
+
+use crate::{connector::docker_run, catalog::{fetch_live_specs, List, SpecTypeSelector, NameSelector, collect_specs}, dataplane::{self, fetch_data_plane_access_token}};
+use crate::local_specs;
+use anyhow::anyhow;
+
+#[derive(Debug, clap::Args)]
+#[clap(rename_all = "kebab-case")]
+pub struct SuggestSchema {
+    /// Collection name
+    #[clap(long)]
+    collection: String,
+
+    /// Task name
+    #[clap(long)]
+    task: String,
```

Comment:
It can be either a materialization or a capture, but I think most of the time a capture will be used. I've added some comments to this command, let me know if you think this could still be improved